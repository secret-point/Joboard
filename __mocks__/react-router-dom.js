// Mock navigate and useParams when needed
// const navigate = jest.fn();
// const useParams = jest.fn();
const useLocation = jest.fn();

module.exports = {
  ...jest.requireActual("react-router-dom"),
  // useNavigate: () => navigate,
  // useParams,
  useLocation,
};
