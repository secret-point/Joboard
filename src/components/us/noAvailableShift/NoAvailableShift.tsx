import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import React, { useEffect } from "react";
import { connect } from "react-redux";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  redirectToDashboard,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { translate as t } from "../../../utils/translator";
import { useLocation } from "react-router";
import queryString from "query-string";
import { useBreakpoints } from "@amzn/stencil-react-components/responsive";
import { GetScheduleListByJobIdRequest } from "../../../utils/apiTypes";
import { checkAndBoundGetApplication, getLocale } from "../../../utils/helper";
import { boundGetScheduleListByJobId } from "../../../actions/ScheduleActions/boundScheduleActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { JobState } from "../../../reducers/job.reducer";
import { ApplicationState } from "../../../reducers/application.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { CandidateState } from "../../../reducers/candidate.reducer";

interface MapStateToProps {
  job: JobState,
  application: ApplicationState,
  candidate: CandidateState
}

const NoAvailableShift = (props: MapStateToProps) => {

  const { job, application, candidate } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId } = queryParams;
  const jobDetail = job.results;
  const applicationData = application.results;
  const candidateData = candidate.results.candidateData;

  useEffect(() => {
    boundGetCandidateInfo();
  },[])

  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() })
  }, [jobDetail, jobId]);

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    // Page will emit page load event once both pros are available but
    // will not emit new event on props change once it has emitted pageload event previously
    jobDetail && applicationData && candidateData && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, candidateData]);

  useEffect(() => {
    return () => {
      //reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    }
  },[])

  const handleGoToDashboard = () => {
    redirectToDashboard();
  }

  return (
    <Col gridGap="S300" padding={{ top: 'S300' }}>
      <Text fontSize="T400">
        {t("BB-no-available-shift-all-shifts-filled-text", "Currently all shifts have been filled.")}
      </Text>
      <Text fontSize="T200">
        {t("BB-no-available-shift-return-dashboard-to-apply-text", "Please return to the dashboard to apply for other jobs.")}
      </Text>
      <Button variant={ButtonVariant.Primary} onClick={handleGoToDashboard}>
        {t("BB-no-available-shift-return-to-dashboard", "Return to dashboard")}
      </Button>
    </Col >
  )
}

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(NoAvailableShift);
