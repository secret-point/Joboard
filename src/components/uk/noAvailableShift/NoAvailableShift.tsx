import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import React, { useEffect } from "react";
import { connect } from "react-redux";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { translate as t } from "../../../utils/translator";
import { useLocation } from "react-router-dom";
import queryString from "query-string";
import { checkAndBoundGetApplication, getLocale, routeToAppPageWithPath } from "../../../utils/helper";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { JobState } from "../../../reducers/job.reducer";
import { ApplicationState } from "../../../reducers/application.reducer";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { PAGE_ROUTES } from "../../pageRoutes";

interface MapStateToProps {
  job: JobState;
  application: ApplicationState;
  candidate: CandidateState;
}

export const NoAvailableShift = (props: MapStateToProps) => {

  const { job, application, candidate } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId } = queryParams;
  const jobDetail = job.results;
  const applicationData = application.results;
  const { candidateData } = candidate.results;

  useEffect(() => {
    boundGetCandidateInfo();
  }, []);

  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobDetail, jobId]);

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    // Page will emit page load event once both pros are available but
    // will not emit new event on props change once it has emitted pageload event previously
    jobDetail && applicationData && candidateData && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, candidateData]);

  useEffect(() => {
    return () => {
      // reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    };
  }, []);

  return (
    <Col gridGap="S300" padding={{ top: "S300" }}>
      <Text fontSize="T400">
        {t("BB-kondo-no-available-shift-all-shifts-filled-text", "More shift are on the way!")}
      </Text>
      <Text fontSize="T200">
        {t("BB-kondo-no-available-shift-return-dashboard-to-apply-text", "We post jobs often so submit your shift preferences and we will contact you when we find a shift that works for you.")}
      </Text>
      <Button
        variant={ButtonVariant.Primary}
        onClick={() => {
          routeToAppPageWithPath(PAGE_ROUTES.SHIFT_PREFERENCE);
        }}
      >
        {t("BB-no-available-shift-submit-shift-preferences", "Submit your shift preferences")}
      </Button>
    </Col>
  );
};

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(NoAvailableShift);
