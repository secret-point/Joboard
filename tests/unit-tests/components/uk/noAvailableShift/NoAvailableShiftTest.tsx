import React from "react";
import { shallow } from "enzyme";
import { useLocation } from "react-router-dom";
import { NoAvailableShift } from "../../../../../src/components/uk/noAvailableShift/NoAvailableShift";
import {
  TEST_APPLICATION_ID,
  TEST_APPLICATION_STATE,
  TEST_CANDIDATE_STATE,
  TEST_JOB_ID,
  TEST_JOB_STATE
} from "../../../../test-utils/test-data";

describe("NoAvailableShift", () => {
  const mockLocation = {
    pathname: "/no-available-shift",
    search: `?jobId=${TEST_JOB_ID}&applicationId=${TEST_APPLICATION_ID}`,
    hash: "",
    state: null
  };
  const mockUseLocation = useLocation as jest.Mock;
  mockUseLocation.mockReturnValue(mockLocation);

  it("should match snapshot", () => {
    const shallowWrapper = shallow(
      <NoAvailableShift
        candidate={TEST_CANDIDATE_STATE}
        job={TEST_JOB_STATE}
        application={TEST_APPLICATION_STATE}
      />);

    expect(shallowWrapper).toMatchSnapshot();
  });
});
