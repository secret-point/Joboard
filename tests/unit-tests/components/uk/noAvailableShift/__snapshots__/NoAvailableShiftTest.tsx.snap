// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoAvailableShift should match snapshot 1`] = `
<Col
  gridGap="S300"
  padding={
    Object {
      "top": "S300",
    }
  }
>
  <Text
    font="primary"
    fontSize="T400"
    hyphens="none"
  >
    More shift are on the way!
  </Text>
  <Text
    font="primary"
    fontSize="T200"
    hyphens="none"
  >
    We post jobs often so submit your shift preferences and we will contact you when we find a shift that works for you.
  </Text>
  <Button
    colorScheme="default"
    iconPosition="leading"
    onClick={[Function]}
    shouldUseResponsiveContent={true}
    size="default"
    type="button"
    variant="primary"
  >
    Submit your shift preferences
  </Button>
</Col>
`;
